csharp_dotnetcore/13.Basic-Bot-Template/Dialogs/Greeting/GreetingDialog.cs

```diff
@@ -56,97 +56,55 @@ public GreetingDialog(BotServices botServices, IStatePropertyAccessor<GreetingSt
-                string[] userNameEntities = { "userName", "userName_paternAny" };
-                string[] userLocationEntities = { "userLocation", "userLocation_patternAny" };
-
-                var entities = turnContext.TurnState[LuisEntities] as JObject;
-                if (entities != null && entities.HasValues)
+                var greetingStateOpt = stepContext.Options as GreetingState;
+                if (greetingStateOpt != null)
                 {
-                    // Update any entities
-                    // Consider having a dialog confirm the update.
-                    foreach (var name in userNameEntities)
-                    {
-                        // check if we found valid slot values in entities returned from LUIS
-                        if (entities[name] != null)
-                        {
-                            greetingState.Name = (string)entities[name][0];
-                            result = true;
-                            break;
-                        }
-                    }
-
-                    foreach (var city in userLocationEntities)
-                    {
-                        if (entities[city] != null)
-                        {
-                            greetingState.City = (string)entities[city][0];
-                            result = true;
-                            break;
-                        }
-                    }
-
-                    // set the new values
-                    await GreetingStateAccessor.SetAsync(turnContext, greetingState);
+                    await GreetingStateAccessor.SetAsync(stepContext.Context, greetingStateOpt);
+                }
+                else
+                {
+                    await GreetingStateAccessor.SetAsync(stepContext.Context, new GreetingState());
                 }
             }
 
-            return result;
-        }
-
-        private async Task<DialogTurnResult> InitializeStateStepAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
-        {
-            // Check to see if there are entities that require updating.
-            await OnUpdateEntitiesAsync(stepContext.Context, cancellationToken);
-
-            var greetingState = await GreetingStateAccessor.GetAsync(stepContext.Context, () => new GreetingState());
-
-            stepContext.Values[NameValue] = greetingState.Name;
-            stepContext.Values[CityValue] = greetingState.City;
-
             return await stepContext.NextAsync();
         }
 
         private async Task<DialogTurnResult> PromptForNameStepAsync(
                                                 WaterfallStepContext stepContext,
                                                 CancellationToken cancellationToken)
         {
-            // Prompt for name if missing
-            if (string.IsNullOrWhiteSpace((string)stepContext.Values[NameValue]))
+            var greetingState = await GreetingStateAccessor.GetAsync(stepContext.Context);
```

Comment:
We renamed greetingState to userProfile on the nodejs side since that reads better. Can you rename this to be userProfile and userProfileAccessor instead of greetingState and greetingStateAccessor?